pkg/compactor/retention/expiration.go

```diff
@@ -132,6 +133,43 @@ func NewTenantsRetention(l Limits) *TenantsRetention {
 	}
 }
 
+func (tr *TenantsRetention) PolicyFor(userID string, lbs labels.Labels) string {
+	policyStreamMapping := tr.limits.PoliciesStreamMapping(userID)
+	var (
+		matchedRule       *validation.PriorityStream
+		matchedPolicyName string
+		found             bool
+	)
+Outer:
+	for policyName, policyStreams := range policyStreamMapping {
```

Comment:
This loop is pretty similar to the `RetentionPeriodFor`. Still I cannot help but to think this is a O(n*m) loop that we do for every single push request we get.

Probably not a big deal if customers configure just a few mappings similarly to stream retention.
As we discussed we may need some caching here, but I'm ok with benchmarking this to see if it's possible or even doing it in a followup PR.